source/src/util/cs_DoubleStackCoroutine.c

```diff
@@ -1,55 +1,56 @@
-#include <cfg/cs_StaticConfig.h>
 #include <stddef.h>
 #include <stdint.h>
 #include <util/cs_DoubleStackCoroutine.h>
+#include <logging/cs_CLogger.h>
 
-enum { WORKING = 1, YIELDING_MICROAPP0 = 2, DONE };
+enum { WORKING = 1, YIELDING_COROUTINE = 2, DONE };
+
+stack_params_t* _stackParams = NULL;
 
 // Get stack pointer
 #define getStackPointer(p) asm volatile("mov %0, sp" : "=r"(p) : :)
 
 // Set stack pointer
 #define setStackPointer(p) asm volatile("mov sp, %0" : : "r"(p))
 
-// We hardcode the stack at the end of the RAM dedicated to the microapp
-stack_params_t* const stackParams = (stack_params_t*)(uintptr_t)(g_RAM_MICROAPP_END);
```

Comment:
I really recommend the solution in https://github.com/crownstone/bluenet/compare/master...microapp/fix-coroutine. I don't know where `_stackParams` ends up and I don't know if the microapp will have access to it. 

This is not correct though:
```
stack_params_t* const stackParams = (stack_params_t*)(uintptr_t)(g_RAM_MICROAPP_END);
```

It should be

```
stack_params_t* const stackParams = (stack_params_t*)(uintptr_t)(g_RAM_MICROAPP_END) - 1;
```

See https://github.com/crownstone/bluenet/compare/master...microapp/fix-coroutine

Reply:
I tried to actually decouple the coroutine from microapps (its not in the microapp dir after all), so the ram end is a parameter. See `startCoroutine` implementation.

Reply:
Ah, check.